tests/DependencyInjection/Compiler/AddDependencyCallsCompilerPassTest.php

```diff
@@ -255,8 +257,10 @@ public function testProcessSortAdmins(): void
         $compilerPass->process($container);
         $container->compile();
 
+        $pool = $container->get('sonata.admin.pool');
+        $this->assertInstanceOf(Pool::class, $pool);
```

Comment:
This would not be needed if using the container.xml definition

Reply:
It's not the same path if we're using sf4 or sf5.
```
        # or with Symfony 4.2+
        container_xml_path: var/cache/dev/srcApp_KernelDevDebugContainer.xml
        # or with Symfony 5+
        container_xml_path: var/cache/dev/App_KernelDevDebugContainer.xml
```
And I have no such file currently...

Reply:
The file is generated when run anything that warms cache, like running tests, the route shouldn't change because we are not really running a Sf app, and we didn't change the directory structure between sf versions. To use it we need a stable cache directory (i think we are using a temp directory), so it would need further refactor.

If it is too hard just forget about it, it can be changed later.

Reply:
This pr is here for a long time and is big enough imho.
I'd would prefer to delay this in another pr.

And it'will work for phpstan but not sure it works for psalm

Reply:
Fine by me, lets do this on other PR. FYI this feature is also for psalm symfony plugin. I think I can try to make it work later, on another pr.